### ISSUE TITLE
High rate of Goodbye logs / messages

### ISSUE THREAD

**Issue by prestonvanloon** [prysmaticlabs/prysm#5796]
# 🐞 Bug Report

### Description

@protolambda observed a high rate of Goodbye message logs at debug level https://github.com/prysmaticlabs/prysm/pull/5589#issuecomment-626233150

Based on my observations, it looks like we might be sending multiple goodbye messages per peer. 

![Screenshot from 2020-05-09 14-42-36](https://user-images.githubusercontent.com/7246818/81485789-b1e01a80-9204-11ea-9176-3aa4137567f4.png)

[Prylabs internal logs](https://kibana.prylabs.network/goto/873081efd6a6753438a9e8f253e6fd1a)

### Has this worked before in a previous version?

The bug was introduced in #5589.

## 🔬 Minimal Reproduction

Run a beacon node with debug logging on.

## 🔥 Error

N/A

## 🌍  Your Environment

**Operating System:**

N/A

**What version of Prysm are you running? (Which release)**

Any after #5589 

**Anything else relevant (validator index / public key)?**

**Comment by protolambda** [comment]
Copying relevant Discord conversation:
```
Debug level logs, got me this:

sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep -o -E "peer=.{53}" | sort | uniq | wc -l
891

891 unique peer ids seen on topaz for the last 4 days
4 days of running in debug mode:

$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep "Goodbye" | wc -l
780398
$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep "Sending Goodbye message to peer\" Reason=\"fault/error" | wc -l
780509
$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep "Goodbye" | wc -l
780599
$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | wc -l
1889091


That's a lot of log messages...
And yes, between running those commands, Goodbye RPC error lines increased by a few hundred
protoToday at 11:35 PM
docker image 9a7fbadca570, commit 2eac24cb79b0c847a74e3f8395ab5808c4059dce
@prestonvanloon was the connection-attempt/goodbye-rpc spam here as well, if you prefer discord over github for chat
prestonvanloonToday at 11:44 PM
hi
i am not 100% sure, what i am sseing in the logs is that we are logging multiple "Sending Goodbye message to peer" per peer ID per node
sometimes as much as 50+ in a short time interval
i am leaning more on the side of "i am spamming someone" rather than "someone is spamming me"
@proto i think we need to use sync.Once or something, it really feels like the callback to send Goodbye is being sent multiple times
I'll compile a new issue for Nishant to look at when he's available
protoToday at 11:48 PM

Example line:
time="2020-05-09 21:37:24" level=debug msg="Sending Goodbye message to peer" Reason="fault/error" peer=16Uiu2HAkvobXD4XYCpuzfXb2z7bEAB4GvXXeTbaEf5NXQR6YXq9w prefix=sync

$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep "Sending Goodbye message to peer\" Reason=\"fault/error" | wc -l
784068
$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep "Sending Goodbye message to peer\" Reason=\"fault/error" | uniq | wc -l
637455
$ sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | grep "Sending Goodbye message to peer\" Reason=\"fault/error" | sort | uniq | wc -l
838

prestonvanloonToday at 11:49 PM
only 836 unique?
protoToday at 11:49 PM
There are definitely consecutive bursts of equal log messages.
But yes, only 836 unique (timestamp stripped off)
prestonvanloonToday at 11:49 PM
gotcha
yeah we have seen this before where Prysm would send like 50 status messages on connection
protoToday at 11:50 PM
And then when not sorted, uniq gets only consecutive duplicates
prestonvanloonToday at 11:50 PM
hopefully won't be too hard to fix, thanks for reporting, i don't think anyone had noticed since it's buried in debug log
protoToday at 11:51 PM
yes, had to check my node since testnode stopped reporting to eth2stats, and other two topaz nodes are not syncing anymore. Still need to check those, but loaded with work

sudo docker logs 8cb989bbd69b 2>&1 | cut -c34- | uniq | grep "Sending Goodbye message to peer\" Reason=\"fault/error" | wc -l
740167

40k less lines when removing pure (the very next line) duplicates
So yes, fully confirm it's either sending duplicates a lot, or connection problems follow after eachother very shortly
prestonvanloonToday at 11:54 PM
ok great, yeah my pi died in the last few days too
and users have been reporting higher resource usage in alpha.6
could be related
```

**Comment by 0xKiwi** [comment]
Update on this @nisdas ? Has the issue been solved with #5799?

**Comment by prestonvanloon** [comment]
I think this was fixed with #5799, github just didn't pick up on the "fixes" keyword with two issues listed.
